lib/hooks.js

```diff
@@ -52,9 +52,11 @@ function Hooks () {
   this.preClose = []
 }
 
+Hooks.property = Object.create(null)
+
 Hooks.prototype.validate = function (hook, fn) {
   if (typeof hook !== 'string') throw new FST_ERR_HOOK_INVALID_TYPE()
-  if (supportedHooks.indexOf(hook) === -1) {
+  if (Array.isArray(this[hook]) === false) {
```

Comment:
Too relax for the check. I would put the supported hook as the static property of `Hooks` class.
So it own `Hooks.ApplicationHooks`, `Hooks.LifeCycleHooks` and `Hooks.CustomHooks`.

Then, it should be extendible in future by adding custom hook name to the `Hooks.CustomHooks`